### PR TITLE
fix mpi test

### PR DESCRIPTION
**Pull request by PhilipVinc**
This PR fixes two things: 
- an mpi test broken by recent jax changes (how they split rng keys) 
- The fact that until https://github.com/google/jax/issues/11916 is fixed, some tests will be failing on recent versions of jax. As the failing tests are quite exotic, this should not be a big problem

### CODE REVIEW

File: test/mpi/test_mpi_utils.py [hunk 1]
```diff
@@ -33,5 +34,14 @@ def test_key_split(_mpi_size, _mpi_comm):
 
     key, _ = nk.jax.mpi_split(key)
     keys = MPI.COMM_WORLD.allgather(key)
-    assert all([not jnp.all(k == keys) for k in keys])
```
Comment (gcarleo) [inline test/mpi/test_mpi_utils.py:36]: why this test was passing before?

Comment (PhilipVinc) [reply test/mpi/test_mpi_utils.py:36]: because it was correct.
but rng-split keys changed type in recent jax versions.

Comment (gcarleo) [reply test/mpi/test_mpi_utils.py:36]: I thought this was checking if the keys are different on different nodes, maybe I misunderstand the test, but it seems to me the old one was checking that after splitting the key is different, even on the same node

Comment (PhilipVinc) [reply test/mpi/test_mpi_utils.py:36]: it was only  checking that the key on a node is not equal to the original list of keys.

I just added a check that it also checks that the key on this node is equivalent to the key in the original array.

**Comment by PhilipVinc** [comment]
Failure is a regression caused by recent changes in jax. I opened an issue in google/jax#11916

**Comment by PhilipVinc** [comment]
This is necessary to fix all other failing PRs

File: netket/vqs/mc/mc_state/expect.py [hunk 2]
```diff
@@ -126,13 +125,19 @@ def logpsi(w, σ):
...
+    # _, Ō_stats = nkjax.expect(
+    #    log_pdf,
+    #    partial(local_value_kernel, logpsi),
+    #    parameters,
+    #    σ,
+    #    local_value_args,
+    #    n_chains=σ_shape[0],
+    # )
+
+    L_σ = local_value_kernel(logpsi, parameters, σ, local_value_args)
```
Comment (gcarleo) [inline netket/vqs/mc/mc_state/expect.py:139]: I actually find this clearer than the previous version, can we leave this? :) or is there a performance hit?

**Review by gcarleo — APPROVED** [review]

File: netket/vqs/mc/mc_state/expect.py (hunk 2 above)
Comment (PhilipVinc) [reply netket/vqs/mc/mc_state/expect.py:139]: They are equivalent now (minus jax lost features) as long as you don't compute gradients of this code (which we never do, as we rewrite it in expect and grad).

If you compute the gradient you need to use expect otherwise you get the wrong gradient.

Originally I wanted to avoid writing this code twice but this never happened so we can keep this.
